packages/core/admin/admin/src/content-manager/pages/EditView/components/DocumentActions.tsx

```diff
@@ -80,57 +104,324 @@ const DocumentActions = ({ actions }: DocumentActionsProps) => {
-              paddingTop="7px"
-              paddingLeft="9px"
-              paddingRight="9px"
-              variant="secondary"
-            >
-              <More aria-hidden focusable={false} />
-              <VisuallyHidden as="span">
-                {formatMessage({
-                  id: 'content-manager.containers.edit.panels.default.more-actions',
-                  defaultMessage: 'More actions',
-                })}
-              </VisuallyHidden>
-            </Menu.Trigger>
-          </Menu.Root>
-        ) : null}
+        <DocumentActionButton {...primaryAction} variant={primaryAction.variant || 'default'} />
+        {restActions.length > 0 ? <DocumentActionsMenu actions={restActions} /> : null}
       </Flex>
       {secondaryAction ? (
-        <Button
-          startIcon={secondaryAction.icon}
-          disabled={secondaryAction.disabled}
-          onClick={secondaryAction.onClick}
-          justifyContent="center"
-          variant="secondary"
-        >
-          {secondaryAction.label}
-        </Button>
+        <DocumentActionButton
+          {...secondaryAction}
+          variant={secondaryAction.variant || 'secondary'}
+        />
       ) : null}
     </Flex>
   );
 };
 
+/* -------------------------------------------------------------------------------------------------
+ * DocumentActionButton
+ * -----------------------------------------------------------------------------------------------*/
+
+interface DocumentActionButtonProps extends Action {}
+
+const DocumentActionButton = (action: DocumentActionButtonProps) => {
+  const [dialogId, setDialogId] = React.useState<string | null>(null);
+  const toggleNotification = useNotification();
+
+  const handleClick = (action: Action) => (e: React.MouseEvent) => {
+    if (action.onClick) {
+      action.onClick(e);
+    }
+
+    if (action.dialog) {
+      switch (action.dialog.type) {
+        case 'notifcation':
+          toggleNotification({
+            title: action.dialog.title,
+            message: action.dialog.content,
+            type: action.dialog.status,
+            timeout: action.dialog.timeout,
+            onClose: action.dialog.onClose,
+          });
+          break;
+        case 'dialog':
+        case 'modal':
+          e.preventDefault();
+          setDialogId(action.id);
+      }
+    }
+  };
+
+  const handleClose = () => {
+    setDialogId(null);
+  };
+
+  return (
+    <>
+      <Button
+        flex={1}
+        startIcon={action.icon}
+        disabled={action.disabled}
+        onClick={handleClick(action)}
+        justifyContent="center"
+        variant={action.variant || 'default'}
+      >
+        {action.label}
+      </Button>
+      {action.dialog && action.dialog.type === 'dialog' ? (
+        <DocumentActionConfirmDialog
+          {...action.dialog}
+          variant={action.variant}
+          isOpen={dialogId === action.id}
+          onClose={handleClose}
+        />
+      ) : null}
```

Comment:
Because i've hit this pitfall taken from the react-docs:

```shell
Don’t put numbers on the left side of &&.

To test the condition, JavaScript converts the left side to a boolean automatically. However, if the left side is 0, then the whole expression gets that value (0), and React will happily render 0 rather than nothing.

For example, a common mistake is to write code like messageCount && <p>New messages</p>. It’s easy to assume that it renders nothing when messageCount is 0, but it really renders the 0 itself!

To fix it, make the left side a boolean: messageCount > 0 && <p>New messages</p>.
```

So i've just taught myself it wasn't worth it 😄 just a preference!